pkg/providers/pricing/suite_test.go

```diff
@@ -82,14 +83,18 @@ var _ = AfterEach(func() {
 var _ = Describe("Pricing", func() {
 	It("should return static on-demand data if pricing API fails", func() {
 		awsEnv.PricingAPI.NextError.Set(fmt.Errorf("failed"))
-		p := pricing.NewProvider(ctx, awsEnv.PricingAPI, awsEnv.EC2API, "", make(chan struct{}))
+		p := pricing.NewProvider(ctx, awsEnv.PricingAPI, awsEnv.EC2API, "")
+		controller := pricing.NewController(p)
```

Comment:
Is there a reason to re-create the controller every time? Why not create the controller once and just reset the data when you move between tests?

Reply:
We create a new provider for each test, and I don't think it would be a good idea to expose switching out the pricing provider. That's why I just made a new controller for each test.

Reply:
If I recall, I think we re-created the provider every time since we didn't have control over the async logic. I don't think we need this anymore now that it's controller based

Reply:
My assumption is that we could just reset the data inside the provider and then re-reconcile the controller when we need to update the pricing at the beginning of the test